### DISCUSSION TITLE
Issue with the `ring-opacity-10` class not taking effect

### DISCUSSION THREAD

**Discussion by onmax** [tailwindlabs/tailwindcss#11565] [Help]
Hello Tailwind CSS team,

I hope you're all doing well. I've recently come across a puzzling problem with the `ring-opacity-10` class, and I'm looking for your help to understand it better.

To illustrate the problem, I've created a [Tailwind Play example](https://play.tailwindcss.com/aC7FCHOuiS). In this example, I am using the `ring-opacity-10` class, which is expected to generate the following CSS:

```css
.ring-opacity-10 {
  --tw-ring-opacity: 0.1;
}
```

However, when I inspect the generated code in the [Tailwind Play example](https://play.tailwindcss.com/aC7FCHOuiS), it appears that the `--tw-ring-opacity` variable isn't actually being used.

I have also checked the documentation and found that it recommends using the `ring-COLOR/opacity` shorthand. This shortcut works perfectly, but does not satisfy my use case. I'd like to use an arbitrary colour (defined by a CSS variable) combined with an opacity level. Ideally I would like to do something like `ring-[--my-colour]/10`, but this doesn't seem to work. I have therefore tried using the `ring-opacity-XX` class as a workaround, but to no avail.

So my questions are

1. Is there a specific reason why Tailwind CSS generates the `ring-opacity-10` class even though it doesn't seem to be used? Am I possibly misusing it?
2. Is there an alternative way to combine an arbitrary colour (defined by a CSS variable) with an opacity level without using the `ring-COLOR/opacity` shorthand?

Thank you! 🙌

**Comment by wongjn** [comment]
1. Tailwind CSS generates the `ring-opacity-10` rule because it is a class candidate that matches a core utility. It does not have compatibility with CSS variable value since there is no way for Tailwind to know the value of the CSS variable at build time and so it can not modify the value in a way to be able to inject the opacity value into the color safely.
2. Have the opacity level be part of the arbitrary value CSS variable, like:
   ```html
   <div class="ring-[--color]" style="--color:rgb(0  0  0 / 0.9)">…</div>
   Or
   <div class="ring-[--color] ring-opacity-10" style="--color:rgb(0  0  0 / var(--tw-ring-opacity))">…</div>
   Or
   <div class="ring-[color-mix(in_srgb,var(--color)_90%,transparent)]">…</div>
   ```

**Reply by onmax** [reply]
Ok, but then I think it is confusing that `ring-opacity-10` is part of the core utility and yet it does nothing unless you explicitly use it with `var()`. Wouldn't be better remove the `ring-opacity-10` as a utility and avoid unnecessary confusion?

Thanks for the answer.

**Reply by wongjn** [reply]
`ring-opacity-*` is for backwards-compatibility when using `ring-<color>` classes that can have their alpha channel value modified, like `ring-red-500` or `ring-[#123]`.
